### ISSUE TITLE
rospkg  identifies macOS sierra and later as "OS X"

### ISSUE THREAD

**Issue by rotu** [ros-infrastructure/rospkg#209]
Internally and in stack traces, rospkg identifies macOS sierra as "OS X". Its official name is macOS

**Comment by dirk-thomas** [comment]
When the logic was implemented the official name was "OS X". Please consider contributing a pull request to update it.

**Comment by erickisos** [comment]
Hey everyone!

I just want to know what's the difference of being detected as OSX vs macOS? I mean, in the functional sense does it have some difference?

**Comment by christophebedard** [comment]
> I mean, in the functional sense does it have some difference?

I don't think so, but it does need to be "osx" for other things to work. The rosdep lists, for example: https://github.com/ros/rosdistro/blob/master/rosdep/osx-homebrew.yaml. So if we were to change it to "macos" we'd need to change it everywhere.

**Comment by erickisos** [comment]
So, we can keep this as OSX but first, we need to be sure that this is working on macOS sierra and higher, that's right?

I mean, i saw the #210 issue and I just want to be sure if it's related based on the id of OS, but it sounds like a different beast to me (related, but different).

**Comment by CursedRock17** [comment]
If it currently works as valid naming conventions, should this issue be closed?

**Comment by christophebedard** [comment]
From what I understand, this issue is not about supporting newer versions. It's about the name of the OS being "macOS" now and not "OS X" like `rospkg` reports
